benchmark/infrastructure/BenchmarkToString.h

```diff
@@ -0,0 +1,48 @@
+// Copyright 2023, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Andre Schlegel February of 2023, schlegea@informatik.uni-freiburg.de)
+
+#pragma once
+
+#include <algorithm>
+#include <sstream>
+#include <string_view>
+#include <vector>
+
+#include "../benchmark/infrastructure/Benchmark.h"
+#include "../benchmark/infrastructure/BenchmarkMeasurementContainer.h"
+#include "BenchmarkMetadata.h"
+#include "util/json.h"
+
+namespace ad_benchmark {
+
+// How the indentation should look like.
+static constexpr std::string_view outputIndentation = "    ";
+
+/*
+@brief Adds indentation before the given string and directly after new line
+characters.
+
+@param indentationLevel How deep is the indentation? `0` is no indentation.
+*/
+std::string addIndentation(const std::string_view str,
+                           const size_t& indentationLevel);
```

Comment:
Wouldn't `size_t indentationLevel` lead to a worse performance? Additionally: I just had the thought, that, maybe, I should move this function to `StringUtils.h`. What do you think?

Reply:
No. Small objects (one or two machine words) are in practice even more efficient when copied by value.
And since this function is in a header, the compiler can do whatever it wants.
But I agree, maybe you can move this to `StringUtils.h` and then also add unit tests for it.

Reply:
Done, see [f261c3f](https://github.com/ad-freiburg/qlever/pull/1010/commits/f261c3f76d5a2ba7a5e11670808ddbc339e845aa).

Can you think of any more unit tests?